### ISSUE TITLE
Truncated USS file downloaded 

### ISSUE THREAD

**Issue by plavjanik** [zowe/zowe-cli#1170]
I am downloading a binary file using the following command:

```sh
zowe zos-files dl uf /a/mydir/sca.jar -f sca.jar --binary
```

The download seems to be successful:

```
USS file downloaded successfully.
Destination: sca.jar
```

But the file is corrupted. It has been created using `jar` tool:

```sh
cd /usr/lpp/zosmf/installableApps
jar cvf /a/plape03/sca.jar izuCA.ear
```

But it cannot be opened using `jar` nor `unzip` on my computer.

When I have used `zftp` plugin to download it, I am getting a different size of the file:
```
-rw-r--r--   1 myuser  staff   9282652 Oct 19 13:24 sca.jar
-rw-r--r--   1 myuser  staff   9291928 Oct 19 11:59 sca3.jar
```

and the FTP transferred file is ok.

They seem to differ only at the end:
![image](https://user-images.githubusercontent.com/7644758/137901081-427a33ed-3ba5-4dcd-bdd0-63e03d127af9.png)

**Comment by t1m0thyj** [comment]
I've been able to reproduce this issue when downloading binary data sets or USS files thru z/OSMF.

The issue is not specific to Zowe CLI as cURL also truncates the end of the file. It only happens when the "Accept-Encoding: gzip" header is passed and the REST response is compressed.

We plan to file an issue with z/OSMF about this, and in the interim to disable gzip compression in Zowe CLI for binary downloads.

**Comment by plavjanik** [comment]
Thank you for working on a workaround and reporting it to IBM.

Would not it be possible to activate the workaround based on the z/OSMF version in case that only some versions have this issue?

**Comment by ATorrise** [comment]
It appears that a couple of viable suggestions were made and no recent activity has occurred, so this issue was closed.

**Comment by t1m0thyj** [comment]
Sorry for not posting an update on this issue in so long. We have an ongoing case open with the z/OSMF team about this, and they are looking into it.

We have reproduced the issue with the following steps:
* Create a 4KB file with random contents: `dd if=/dev/urandom of=testfile bs=1024 count=4`
* Upload the file to a sequential data set: `zowe files upload ftds testfile <dsname> --binary`
* Download the file with cURL
  * Uncompressed: `curl -H "X-IBM-Data-Type: binary" "<zosmfUrl>/zosmf/restfiles/ds/<dsname>"` 
  * Compressed: pass the "Accept-Encoding: gzip" header and `--compressed` option
* Notice that the downloaded file is truncated only when gzip compression is enabled

I've added a "Blocked" label for now. Once the issue with z/OSMF is resolved, we can reenable gzip compression in Zowe CLI for binary downloads.

**Comment by t1m0thyj** [comment]
> Would not it be possible to activate the workaround based on the z/OSMF version in case that only some versions have this issue?

I don't believe this is a version-specific bug. Gzip compression support is a fairly recent addition to z/OSMF so I think this bug is present in all versions of z/OSMF that support gzip compression.
